language-plutus-core/src/Language/PlutusCore/Renamer.hs

```diff
@@ -112,50 +115,71 @@ annotateTy (TyApp x ty ty') =
 annotateTy (TyBuiltin x tyb) = pure (TyBuiltin x tyb)
 annotateTy (TyInt x n) = pure (TyInt x n)
 
--- | Mapping from locally unique indices to globally unique 'Unique's.
-newtype UniquesRenaming = UniquesRenaming
-    { unUniquesRenaming :: IM.IntMap Unique
+-- | Mapping from locally unique indices to globally unique uniques.
+data ScopedUniquesRenaming = ScopedUniquesRenaming
+    { _scopedUniquesRenamingTypes :: IM.IntMap TypeUnique
+    , _scopedUniquesRenamingTerms :: IM.IntMap TermUnique
     }
 
+makeLenses ''ScopedUniquesRenaming
+
+class Coercible Unique unique => HasUniquesRenaming unique where
```

Comment:
The `Coercible` constraint here doesn't actually do anything, does it?

Reply:
I guess you're effectively using it to bring in the constraint at the use site.

Reply:
> I guess you're effectively using it to bring in the constraint at the use site.

Yeah. Well, a `unique` is supposed to be a `Unique` in the end, so the constraint is sane.